lib/pages/authen/register/widgets/register_button.dart

```diff
@@ -0,0 +1,56 @@
+import 'package:flutter/material.dart';
+import 'package:structure_flutter/bloc/bloc.dart';
+import 'package:structure_flutter/core/resource/app_colors.dart';
+import 'package:structure_flutter/core/resource/text_style.dart';
+
+class RegisterButton extends StatelessWidget {
+  final String _imageURL;
+
+  final RegisterBloc _registerBloc;
+
+  final TextEditingController _emailController;
+
+  final TextEditingController _passwordController;
+
+  final TextEditingController _fullNameController;
+
+  RegisterButton(
+    this._registerBloc,
+    this._fullNameController,
+    this._emailController,
+    this._passwordController,
+    this._imageURL,
+  );
+
+  @override
+  Widget build(BuildContext context) {
+    return Container(
+      height: 35,
+      margin: EdgeInsets.all(20.0),
+      decoration: BoxDecoration(
+        borderRadius: BorderRadius.all(Radius.circular(5.0)),
+        gradient: LinearGradient(colors: [
+          AppColors.blueAccentColor,
+          AppColors.lightBlueAccentColor
+        ]),
+        color: AppColors.lightBlueColor,
+      ),
+      child: FlatButton(
+        onPressed: () => registerSubmitted(),
+        child: Text(
+          'REGISTER',
+          style: AppStyles.white_bold_11,
+        ),
+      ),
+    );
+  }
+
+  void registerSubmitted() {
+    _registerBloc.add(RegisterWithCredentials(
```

Comment:
should not pass a bloc into a widget view like that. try to pass an action instead of bloc

Reply:
I think we should add event with the  same widget because it's make we can maintain easier !

Reply:
thank you for your suggestion ! I will fix this problem right now :]]